### ISSUE TITLE
valid

### ISSUE THREAD

**Issue by fantuan812** [bdi-lab/InGram#10]
Hi, thanks for sharing your codes. In the verification process, msg is passed into the model to generate and embed to calculate the ranking, so that the information of the verification set is leaked to the model, so that the model does not know what entity has what relationship. This link prediction is not a bit of a problem

**Comment by jaejunlee714** [comment]
Hi, thanks for your interest in our paper!
I'm not sure about the verification process you mentioned. Dud you mean the validation process?
If so, we only check the model performance, and the loss is not calculated on the validation set. Also, the triplets in validation set are diffrent from those in the message set. Therefore, the model does not learn the information in the validation set, and there is no leakage of the validation set to the model. If I misunderstood your question, could you kindly clarify it?

Best,
Jaejun Lee

**Comment by fantuan812** [comment]
Thanks for your reply. Before, in the validation process, I passed all the validation sets into the model as information sets, so the experimental effect is good, but there seems to be information leakage. I then tried to use the training information set in the validation session, but it didn't work well.I am unsure whether this approach is correct.

**Comment by jaejunlee714** [comment]
Our datasets have four splits: train, msg, valid, test. The entities and (some) relations in train differ from those in msg, valid, and test. For link prediction on valid and test which have different entities and relations from the train set, we need some auxiliary information, which is the msg. The triplets in msg provides some information about the entities and relations in valid and test set, while having different triplets. Therefore, valid/test requires msg.

**Comment by fantuan812** [comment]
Thank you for your response. I now understand the relationship between the message and validation sets. I would also like to inquire whether the message is necessary if both my validation set and training set contain identical relationships and entities

**Comment by jaejunlee714** [comment]
If the validation set and the training set contain identical entities and relations, then you can use the train set as the message set.

**Comment by fantuan812** [comment]
Thanks for your advice, I will try these. Good luck with your research
